cmd/operator-sdk/add/api.go

```diff
@@ -139,23 +140,30 @@ func apiRun(cmd *cobra.Command, args []string) error {
 			return err
 		}
 	case projutil.OperatorTypeHelm:
-		absProjectPath := projutil.MustGetwd()
-		projectName := filepath.Base(absProjectPath)
-		cfg := input.Config{
-			AbsProjectPath: absProjectPath,
-			ProjectName:    projectName,
-		}
-		createOpts := helm.CreateChartOptions{
-			ResourceAPIVersion: apiFlags.APIVersion,
-			ResourceKind:       apiFlags.Kind,
-			Chart:              apiFlags.HelmChartRef,
-			Version:            apiFlags.HelmChartVersion,
-			Repo:               apiFlags.HelmChartRepo,
-			CRDVersion:         apiFlags.CrdVersion,
-		}
-		if err := helm.API(cfg, createOpts); err != nil {
-			return err
+		helmPlugin := helm.Plugin{}
+		helmCreateAPIPlugin := helmPlugin.GetCreateAPIPlugin()
+
+		// The new command ask for APIVersion (e.g app.example.com/v1alpha1)
+		// Where; app == group , example.com == domain, v1alpha1 == version
+		// However, inside of the current implementation we also will break as requested by the new layout
+		// So, we are here parsing the value and adding flags for the plugin
+		api := strings.Split(apiFlags.APIVersion, "/")
+		version := api[1]
+		group := strings.Split(api[0], ".")[0]
+		domain := string(api[0][len(group)+1 : len(api[0])])
```

Comment:
same here. we'd better assume the string is not of the expected format.

Reply:
Ditto.